ledger-api/grpc-definitions/com/daml/ledger/api/v1/commands.proto

```diff
@@ -125,6 +125,10 @@ message Commands {
   // Additional contracts used to resolve contract & contract key lookups.
   // Optional
   repeated DisclosedContract disclosed_contracts = 17;
+
+  // Must be a valid domain ID
+  // Optional
```

Comment:
So this would take precedence over the workflow_id ?

Reply:
Yes, moreover the hack could be removed too. (if not too much noise wrt testing)

Reply:
Yup, if we can remove the hack, that would be great. That would break some tests but we would happily fix that.

One generic question: do we want to only allow domain ID or to have an enum that could be either a domain ID or a domain alias?

Reply:
as discussed offline for this increment we go with domain_id everywhere except for the connected-domains endpoint.